### PR TITLE
Create note, warning, and info shortcodes

### PR DESCRIPTION
**Pull request by benolayinka**
#### Summary

Create `{{< note />}}`, `{{< warning />}}`, and `{{< info />}}` shortcodes.

image

#### Changes

- Add CSS
- Update CONTRIBUTING and template doc
- Replace all instances of `>Note:` 

#### Notes for Reviewers

@nejraselimovic when you standardized **Note**, did you have some clever method for replacing all? Do you have time to finish up this PR with your magic?

Replace the existing `note` tags with the shortcode -> DONE

#### Checklist

- [x] Scope: The referenced issue is addressed, there are no unrelated changes.
- [x] Run Locally: Verified that the docs build using `make server`
- [ ] New Features Marked: Documentation for new features is marked using the `new-in-version` shortcode, according to the guidelines in [CONTRIBUTING](CONTRIBUTING.md).
- [x] Style Guidelines: Documentation obeys style guidelines in [CONTRIBUTING](CONTRIBUTING.md).
- [x] Commits: Commit messages follow guidelines in [CONTRIBUTING](CONTRIBUTING.md), there are no fixup commits left.

### CODE REVIEW

**Comment by nejraselimovic** [comment]
@benolayinka nothing clever, I just got my hands dirty lol but let me work on replacing these

**Review by htdvisser — CHANGES_REQUESTED** [review]
I'm not a big fan of this. Not every blockquote is a note.

**Comment by johanstokking** [comment]
> I'm not a big fan of this. Not every blockquote is a note.

I agree. If anything, there should be a shortcode instead.

**Comment by htdvisser** [comment]
Exactly. Something like a `note.html` shortcode:

```
<div class="notification is-primary">
  <h3 class="has-text-light">Note</h3>
  {{ .Inner | markdownify }}
</div>
```

that can be used like this:

```
{{< note >}}
This is a note
{{</ note >}}
```

**Comment by kschiffer** [comment]
Indeed let's distinguish this better. I propose styling like this:

1.  Blockquotes -> gray line on the left
2.  Notes -> gray bg, more or less what we have now
3.  Info -> blue bg, white text color
4.  Warnings -> either orange BG, white text color, or orange left line, black text color (see also https://github.com/TheThingsIndustries/lorawan-stack-docs/pull/107#discussion_r526831119)

cc @pierrephz you might want to propose something more concrete even.

**Review by pierrephz — CHANGES_REQUESTED** [review]
I agree we should use boxes for these cases, but I suggest to use some pale colors in the background, with borders. It will make it identifiable from the rest of the content

Here is what I suggest:

image

**Comment by benolayinka** [comment]
> Exactly. Something like a `note.html` shortcode:
> 
> ```
> <div class="notification is-primary">
>   <h3 class="has-text-light">Note</h3>
>   {{ markdownify (or (.Get 0) .Inner) }}
> </div>
> ```
> 
> that can be used like this:
> 
> ```
> {{< note >}}
> This is a note
> {{</ note >}}
> ```
> 
> or like this:
> 
> ```
> {{< note "This is a note" />}}
> ```

This looks good to me. I wanted to minimize the amount of typing required for the 99% case, but an inline note shortcode is short enough and much clearer, you're right.

**Comment by benolayinka** [comment]
> I agree we should use boxes for these cases, but I suggest to use some pale colors in the background, with borders. It will make it identifiable from the rest of the content
> 
> Here is what I suggest:
> 
> image
> 
> cc @benolayinka

@pierrephz @kschiffer how's this? easy to do with Bulma's built in notifications

Screenshot 2020-11-24 at 14 11 25

**Comment by pierrephz** [comment]
@benolayinka cool, small modifications required:
- add borders,
- don't use ITC avant-grade,
- use same font-size for title and body
- use bold font-weight for title

**Comment by nejraselimovic** [comment]
This also needs to be taken care of:

note

**Comment by benolayinka** [comment]
> This also needs to be taken care of:
> 
> note

@htdvisser shortcodes don't work in other shortcodes :/ have an idea?

**Comment by htdvisser** [comment]
I'm pretty sure shortcode-in-shortcode should work just fine with the multiline version. With the oneliner I can indeed imagine that it doesn't work.

**Comment by benolayinka** [comment]
> I'm pretty sure shortcode-in-shortcode should work just fine with the multiline version. With the oneliner I can indeed imagine that it doesn't work.

@nejraselimovic , try

{{< note >}}
This is a note with {{< tts >}} in it
{{</ note >}}

**Comment by nejraselimovic** [comment]
> > I'm pretty sure shortcode-in-shortcode should work just fine with the multiline version. With the oneliner I can indeed imagine that it doesn't work.
> 
> @nejraselimovic , try
> 
> {{< note >}}
> This is a note with {{< tts >}} in it
> {{</ note >}}

@benolayinka All note instances should be replaced now, I went through everything. Don't forget that when there's something under backquotes (like `this`) inside note/info/warning, the background behind it is being rendered to light grey, so this should be fixed too

**Review by htdvisser — APPROVED** [review]

**Comment by benolayinka** [comment]
> > > I'm pretty sure shortcode-in-shortcode should work just fine with the multiline version. With the oneliner I can indeed imagine that it doesn't work.
> > 
> > 
> > @nejraselimovic , try
> > {{< note >}}
> > This is a note with {{< tts >}} in it
> > {{</ note >}}
> 
> @benolayinka All note instances should be replaced now, I went through everything. Don't forget that when there's something under backquotes (like `this`) inside note/info/warning, the background behind it is being rendered to light grey, so this should be fixed too

This is on purpose from Bulma, looks okay to me

Screenshot 2020-11-26 at 19 20 23

**Review by nejraselimovic — APPROVED** [review]

**Comment by benolayinka** [comment]
@pierrephz, @laurensslats voted to just leave the bulma styling as is. i made the appropriate changes and dismissed your review so the pr can move

**Comment by kschiffer** [comment]
@laurensslats can you give some reasoning here? The changes requested are minimal.

**Comment by benolayinka** [comment]
> @laurensslats can you give some reasoning here? The changes requested are minimal.

@laurensslats voted to keep Bulma styling because it matches the rest of the page styles. I pushed for it because I also think it looks good, and it's zero custom css.

I think because @pierrephz has mocked up the entire documentation site, there are differences in fonts, spacing, and heading styles that mean if we actually implement the mockup in this case, it will look out of place. Here's @pierrephz mockup:

image


And here's what we have by default in Bulma:

image

**Comment by kschiffer** [comment]
So are we going to make design adjustments only to the extent that we feel comfortable modifying bulma or how is this going to work? 

This is pretty much exactly the reason why I oppose using ready-made design toolkits such as bulma, because it effectively forces a foreign design on us.

 Then again I realize we don't have the resources to build our own documentation site according to a design specification, so :shrug: ...

**Review by kschiffer — APPROVED** [review]

**Comment by pierrephz** [comment]
> @laurensslats voted to keep Bulma styling because it matches the rest of the page styles. I pushed for it because I also think it looks good, and it's zero custom css.
> 
> I think because @pierrephz has mocked up the entire documentation site, there are differences in fonts, spacing, and heading styles that mean if we actually implement the mockup in this case, it will look out of place. Here's @pierrephz mockup:

Yeah but I kind of disagree with this.

I mocked up the doc for a good reason: make it clean, readable, and consistent with our branding to have something professional

Now it's close to becoming a mess (font size is too big for no reason, every section is stuck to each other, we are using ITC avant garde font where we shouldn't, etc.)

So if we go that way, at some point I will have to take two or three days to go into the code and fix those details

One bad detail + one bad detail + ... and our product is messy and unprofessional

**Comment by benolayinka** [comment]
> > @laurensslats voted to keep Bulma styling because it matches the rest of the page styles. I pushed for it because I also think it looks good, and it's zero custom css.
> > I think because @pierrephz has mocked up the entire documentation site, there are differences in fonts, spacing, and heading styles that mean if we actually implement the mockup in this case, it will look out of place. Here's @pierrephz mockup:
> 
> Yeah but I kind of disagree with this.
> 
> I mocked up the doc for a good reason: make it clean, readable, and consistent with our branding to have something professional
> 
> Now it's close to becoming a mess (font size is too big for no reason, every section is stuck to each other, we are using ITC avant garde font where we shouldn't, etc.)
> 
> So if we go that way, at some point I will have to take two or three days to go into the code and fix those details
> 
> One bad detail + one bad detail + ... and our product is messy and unprofessional

Yep I agree - but there are just some things in the mockup that aren't in the css yet, like the red margins, the new font, and proper section spacing.

Can we compromise on this? It looks much better to me without the colored borders (we dont use borders anywhere else), and I think the title should be at least a tiny bit larger than the text, which matches the section headers as we have them now.

**Comment by pierrephz** [comment]
> Yep I agree - but there are just some things in the mockup that aren't in the css yet, like the red margins, the new font, and proper section spacing.
> 
> Can we compromise on this? It looks much better to me without the colored borders (we dont use borders anywhere else), and I think the title should be at least a tiny bit larger than the text, which matches the section headers as we have them now.

ok, we can compromise: keep it without borders for now, but the title needs a rework;
- font size is not supposed to match the section header and should be reduced
- the box color is already eye-catching, so we should just use the Lato Bold font for titles

@benolayinka Do we agree on that? I don't think its hard to make those fixes

**Comment by benolayinka** [comment]
> > Yep I agree - but there are just some things in the mockup that aren't in the css yet, like the red margins, the new font, and proper section spacing.
> > Can we compromise on this? It looks much better to me without the colored borders (we dont use borders anywhere else), and I think the title should be at least a tiny bit larger than the text, which matches the section headers as we have them now.
> 
> ok, we can compromise: keep it without borders for now, but the title needs a rework;
> 
> * font size is not supposed to match the section header and should be reduced
> * the box color is already eye-catching, so we should just use the Lato Bold font for titles
> 
> @benolayinka Do we agree on that? I don't think its hard to make those fixes

Sweet. How is this?

Screenshot 2020-12-07 at 15 20 19

**Review by pierrephz — APPROVED** [review]

**Comment by pierrephz** [comment]
> Sweet. How is this?
> 
> 
> 
> Screenshot 2020-12-07 at 15 20 19

Cool let's go with this

File: doc/content/devices/adding-devices/_index.md [hunk 1]
```diff
@@ -16,7 +16,7 @@ This section contains instructions for adding devices in {{% tts %}}.
 
 Go to **Devices** in the left menu and click on **+ Add Device** to reach the end device registration page. Fill the device ID, the LoRaWAN MAC and PHY versions and the frequency plan used by the device.
 
->**Note**: The PHY version represents the revision of the LoRAWAN Regional Parameters that the device expects, and must be correlated with the MAC version.
```
Comment (neoaggelos) [inline doc/content/devices/adding-devices/_index.md:19]: Is it `{{< note "message" />}}` or `{{< note >}} message {{</ note >}}`?

**Review by neoaggelos — APPROVED** [review]

File: doc/content/devices/adding-devices/_index.md (hunk 1 above)
Comment (benolayinka) [reply doc/content/devices/adding-devices/_index.md:19]: both work

**Review by adriansmares — APPROVED** [review]
Code owned files LGTM

**Review by KrishnaIyer — APPROVED** [review]
Owned files LGTM

**Pull request merged**